projects/notebooks/visualizations/rhods-notebooks-performance/analyze/__init__.py

```diff
@@ -1,17 +1,60 @@
 import json
 import logging
+import numpy as np
 
-import matrix_benchmarking.common as common
+from typing import List
 
+import matrix_benchmarking.common as common
+import matrix_benchmarking.regression as regression
+import matrix_benchmarking.regression.zscore as zscore
 
 def run():
-    logging.info(f"Received {common.Matrix.count_records()} new entries")
-    for entry in common.Matrix.all_records():
-        pass
 
     logging.info(f"Received {common.LTS_Matrix.count_records()} historic LTS entries")
-    for lts_entry in common.LTS_Matrix.all_records(): pass
+    lts_entries = []
+    # Take the first result if they were gathered
+    for entry in common.LTS_Matrix.all_records():
+        if entry.results and type(entry.results) is list:
+            entry.results = entry.results[0]
+            lts_entries.append(entry)
+        else:
+            lts_entries.append(entry)
+
+    logging.info(f"Received {common.Matrix.count_records()} new entries")
 
     number_of_failures = 0
+    settings_to_check = ["version", "repeat"]
+    for entry in common.Matrix.all_records():
+        regression_results_dest = entry.location / "regression.json"
+        regression_results: List[models.RegressionResult] = []
+        for check_setting in settings_to_check:
+            controlled_settings = entry.get_settings()
+
+            try:
+                controlled_settings.pop(check_setting)
```

Comment:
same as above, you cannot (shouldn't) modify the object given to you, except if you know for sure it's a free copy. if it's not (it's not, see below), or you don't know it
```
    def get_settings(self) -> dict:
        return self.settings.__dict__
```

make a copy yourself
```suggestion
            controlled_settings = dict(entry.get_settings())

            try:
                controlled_settings.pop(check_setting)
```

Reply:
Got it! Didn't really occur to me that `self.settings`  was returning the dict like that. I suppose I've been spoiled by Rust style ownership